iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusTerminal.java

```diff
@@ -82,15 +82,19 @@ public MergedBus getConnectableBus() {
     }
 
     void setConnectableBusId(String connectableBusId) {
-        this.connectableBusId.set(network.get().getVariantIndex(), connectableBusId);
+        int variantIndex = network.get().getVariantIndex();
+        this.connectableBusId.set(variantIndex, connectableBusId);
     }
 
     String getConnectableBusId() {
         return this.connectableBusId.get(network.get().getVariantIndex());
     }
 
     void setConnected(boolean connected) {
-        this.connected.set(network.get().getVariantIndex(), connected);
+        int variantIndex = network.get().getVariantIndex();
+        boolean oldValue = this.connected.set(variantIndex, connected);
+        String variantId = network.get().getVariantManager().getVariantId(variantIndex);
+        getConnectable().notifyUpdate("connected", variantId, oldValue, connectableBusId);
```

Comment:
Replace connectableBusId by connected.

You should probably declare notifyUpdate generic, to ensure that oldValue and newValue have the same types.